### PR TITLE
Widget separator by using extension on iterator

### PR DESCRIPTION
**Pull request by burhanrashid52**
## Description
Usage in this article [How to Create Separators for Rows and Columns](https://widgettricks.substack.com/p/separators-for-rows-columns)

## Basic Checklist

- [x] ✨ Ran `dartdoc`
- [x] 🛠️ Added proper documentation
- [x] ❌ No warnings or errors in the code
- [ ] ✅ If added a new extension, added in Readme and example too
- [x] 🧪 If added a new extension, written proper test cases for it

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by burhanrashid52** [comment]
@AbhishekDoshi26 PR is up. Let me know if you need anything else.

File: README.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ In this package, right now we have the following extension methods:
 - **Extensions on num:**
     - SizedBox hSizedBox
     - SizedBox wSizedBox
+- **Extension on Iterable**
```
Comment (AbhishekDoshi26) [inline README.md:66]: Add a <br> before this.

File: lib/src/iterator_extension.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/widgets.dart';
+
+/// Extension on [Iterable] to add a separator between each widget using generators.
```
Comment (AbhishekDoshi26) [inline lib/src/iterator_extension.dart]: Here, the doc should be just `Extension on Iterable`.

File: lib/src/iterator_extension.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+extension IterableExt on Iterable<Widget> {
+  Iterable<Widget> separator(Widget element) sync* {
```
Comment (AbhishekDoshi26) [inline lib/src/iterator_extension.dart:19]: On the `separator` you should add the doc for separator.

Comment (AbhishekDoshi26) [reply lib/src/iterator_extension.dart:19]: Also, in the doc itself, add an example. For reference, you can check out the snackbar documentation

**Review by AbhishekDoshi26 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (burhanrashid52) [reply README.md:66]: I did not understand this.

Comment (AbhishekDoshi26) [reply README.md:66]: Oh haha! You need to add a `<br>` tag before this line.

Comment (burhanrashid52) [reply README.md:66]: @AbhishekDoshi26 Done

**Review by AbhishekDoshi26 — APPROVED** [review]
Thanks 🚀

**Pull request merged**